packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/index.js

```diff
@@ -69,16 +77,79 @@ const confirmDialogsPropTypes = {
  * ConfirmDialogPublishAll
  * -----------------------------------------------------------------------------------------------*/
 
-const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
+export const ConfirmDialogPublishAll = ({
+  isOpen,
+  onToggleDialog,
+  isConfirmButtonLoading,
+  onConfirm,
+}) => {
   const { formatMessage } = useIntl();
+  const { get } = useFetchClient();
+  const { selectedEntries } = useTableContext();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler(getTrad);
+  const {
+    contentType: { uid: slug },
+  } = useSelector(listViewDomain());
+
+  const {
+    data: countDraftRelations,
+    isLoading,
+    isError,
+  } = useQuery(
+    ['content-manager', 'draft-relations', slug, selectedEntries],
+    async () => {
+      /**
+       * TODO: refactor the API call when the param serializer will be ready
+       */
+      const {
+        data: { data },
+      } = await get(
+        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations?${stringify(
+          { ids: selectedEntries },
+          { encode: false }
+        )}`
+      );
+
+      return data;
+    },
+    {
+      enabled: isOpen,
```

Comment:
What's the goal of delaying the request here? It looks like it could just pre-load in the background, with selectedEntries in the react query key it should keep in sync with the selection anyway

Reply:
That's an interesting idea, if we were to load in the background we probably need to be cancelling the requests?

Reply:
without it I have multiple calls to the API, instead with isOpen it calls the API only once

Reply:
> without it I have multiple calls to the API, instead with isOpen it calls the API only once

This is what we're trying to discuss – why do you think one API call is better than the multiple in comparison to the UX a user will have